predicators/ml_models.py

```diff
@@ -371,25 +375,51 @@ def _create_optimizer(self) -> optim.Optimizer:
         """Create an optimizer after the model is initialized."""
         return optim.Adam(self.parameters(), lr=self._learning_rate)
 
+    def _reset_weights(self) -> None:
+        """(Re-)initialize the network weights."""
+        self.apply(lambda m: self._weight_reset(m, self._weight_init))
+
+    @staticmethod
+    def _weight_reset(m: torch.nn.Module, weight_init: str) -> None:
+        if isinstance(m, nn.Linear):
```

Comment:
what else could this be? `nn.ModuleList`? maybe we should assert that. I'm concerned that one day we'll add, e.g., `nn.Conv2D` somewhere, and we won't realize that the weights aren't getting reset